src/main.rs

```diff
@@ -31,44 +40,48 @@ struct Args {
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
     let args = Args::parse();
-    println!("{:?}", args);
 
-    let market = market::Market::new();
+    let market = Market::new();
     let items: Vec<_> = market
         .fetch_items()
         .await?
         .into_iter()
         .filter(|Item { name, .. }| args.items.contains(name))
         .collect();
 
-    let mut orders: HashMap<&str, Vec<Order>> = HashMap::new();
+    let mut orders = HashMap::<_, Vec<_>>::new();
     for item in &items {
         market
             .fetch_orders(&item.url_id)
             .await?
             .into_iter()
-            .filter(|Order { quantity, platinum_price, user, order_type, .. }| {
+            .filter(|order @ Order { quantity, platinum, user, .. }| {
                 quantity >= &args.quantity
-                    && platinum_price <= &args.max_price
+                    && platinum <= &args.max_price
                     && user.status == "ingame"
-                    && order_type == "sell"
+                    && order.ty == "sell"
             })
             .for_each(|order| {
-                #[rustfmt::skip]
-                orders.entry(&item.name)
-                    .and_modify(|orders| orders.push(order))
-                    .or_default();
+                let Order { user, .. } = &order; // isn't working in param match
+                match orders.entry(user.name.clone()) {
+                    Entry::Occupied(mut entry) => entry.get_mut().push((item, order)),
```

Comment:
Why do we need `get_mut` here?

Reply:
Because this is [`Entry`](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html) type